File: cet-core/src/main/java/com/ckenergy/cet/core/AutoTrackHelper.java
package com.ckenergy.cet.core;

import android.text.TextUtils;
import android.util.Log;

public class AutoTrackHelper {

    private static final String TAG = "AutoTrackHelper";

    public static  void trackViewScreen(String eventName) {
        if (TextUtils.isEmpty(eventName)){
            return;
        }
        Log.d(TAG, "trackViewScreen:"+eventName);
    }

    public static  void trackClick(String eventName) {
        if (TextUtils.isEmpty(eventName)){
            return;
        }
        Log.d(TAG, "trackClick:"+eventName);
    }

}

File: cet-core/src/main/java/com/ckenergy/cet/core/AutoTrackHelper.kt
package com.ckenergy.cet.core

object AutoTrackHelper {
    private const val TAG = "AutoTrackHelper"

    private var eventListener: ComposeEventListener? = null

    fun setEventListener(eventListener: ComposeEventListener) {
        this.eventListener = eventListener
    }

    /**
     * compose page change event
     * @param eventName eg: compose route：ROUTE_MAIN
     */
    @JvmStatic
    fun trackViewScreen(eventName: String?) {
        if (eventName.isNullOrEmpty()) {
            return
        }
        eventListener?.onPageChange(eventName)
    }

    /**
     * compose click event
     * @param eventName eg: com.ckenergy.compose.event.track.MyText.<anonymous>.clickable (MainPage.kt:31)
     * */
    @JvmStatic
    fun trackClick(eventName: String?) {
        if (eventName.isNullOrEmpty()) {
            return
        }
        eventListener?.onClick(eventName)
    }
}